docs/source/spec-workload-executor.rst

```diff
@@ -43,29 +43,98 @@ After accepting the inputs, the workload executor:
-
-   * MUST ignore the ``testData`` array. ``astrolabe`` is responsible for initializing the cluster with
+#. MUST parse the incoming the ``driverWorkload`` document and set up
+   the driver's unified test runner to execute the provided workload, with
+   the following deviations from the unified test runner specification:
+   
+   - Any `MongoClients <https://github.com/mongodb/specifications/blob/master/source/unified-test-format/unified-test-format.rst#entity-client>`_
+     that are instantiated by the workload executor MUST use the input
+     connection string as provided to the workload executor, and MUST
+     apply URI options specified in the particular test, if any, over the
+     provided connection string.
+   
+   - Each MongoClient MUST be set up to publish `command monitoring
+     <https://github.com/mongodb/specifications/blob/master/source/command-monitoring/command-monitoring.rst>`_
+     events. The workload executor MUST record all events published
+     in the course of scenario execution.
+   
+   - Each MongoClient MUST be set up to publish `CMAP
+     <https://github.com/mongodb/specifications/blob/master/source/command-monitoring/command-monitoring.rst>`_
+     events. The workload executor MUST record all events published
+     in the course of scenario execution.
+   
+   - The ``initialData`` array in the scenario MUST be ignored by the
+     unified test runner (and by the workload executor).
+     ``astrolabe`` is responsible for initializing the cluster with
      this data *before* starting the workload executor.
-   * MUST run operations sequentially and in the order in which they appear in the ``operations`` array.
-   * MUST repeat the entire set of specified operations indefinitely, until the **termination signal** from
-     ``astrolabe`` is received.
-   * MUST keep count of the number of operations failures (``numFailures``) that are encountered while running
-     operations. An operation failure is when the actual return value of an operation does not match its
-     expected return value (as defined in the ``result`` field of the ``driverWorkload``).
-   * MUST keep count of the number of operation errors (``numErrors``) that are encountered while running
-     operations. An operation error is when running an operation unexpectedly raises an error. Workload executors
-     implementations should try to be as resilient as possible to these kinds of operation errors.
-   * MUST keep count of the number of operations that are run successfully (``numSuccesses``).
+     
+#. MUST use the driver's unified test runner to execute the tests in the
+   scenario, and the operations in each test, sequentially and in the order
+   in which they appear in the ``tests`` and ``operations`` arrays,
+   with the following deviations from the unified test runner specification:
+   
+   * The workload executor MUST repeat execution of the entire set of
+     specified tests and operations indefinitely, until the
```

Comment:
'specified tests and operations indefinitely' --> what do you mean by specified 'tests'? Isn't the whole thing that is being run a 'test'? I suggest just referring to operations and not tests.

Reply:
Since each test will re-create entities (e.g. new MongoClients performing fresh discovery), I do think you'd need to clarify that the `operations` array is repeated in a loop rather than `tests`.

That said, see my other comment about why a special test operation may be a better way to address the needs of this project without making significant changes to the entire unified test runner.

Reply:
Each test consists of zero or more operations, but the unified runner executes tests, not operations. Specifying that operations must be repeated indefinitely is jumping abstraction layers and descending into unified runner's guts.

Reply:
To clarify, existing DAT structure is scenario -> operations; with unified test format the structure becomes scenario -> tests -> operations.

Reply:
Having the test runner repeat tests (i.e. elements in the `tests` array) would have the side effect of the runner also recreating its entities each time is starts a test. I don't see how you can avoid needing to change the test runner's internals with either approach (telling it to loop on `tests` or execute a single test and loop on `operations`); however, trying to contain all of this logic in a special operation seems like it may help avoid _most_ changes.

Reply:
I see what you mean. There is now a `loop` operation defined in https://github.com/mongodb/specifications/pull/903 that allows using a single test, and the test runner no longer repeats the tests (or operations).

Reply:
@prashantmital: I think this can be resolved now that https://github.com/mongodb/specifications/pull/903 has a `loop` operation.